state-sync/src/executor_proxy.rs

```diff
@@ -217,12 +218,15 @@ impl ExecutorProxyTrait for ExecutorProxy {
             .configs()
             .iter()
             .filter(|(id, cfg)| {
-                &self
-                    .on_chain_configs
-                    .configs()
-                    .get(id)
-                    .expect("missing on-chain config value in local copy")
-                    != cfg
+                if let Some(config) = &self.on_chain_configs.configs().get(id) {
+                    config != cfg
+                } else {
+                    error!(LogSchema::new(LogEntry::Reconfig).error(
+                        &Error::UnexpectedError(format!("Missing on-chain config id: {:?}", id))
+                            .into()
+                    ));
+                    false
+                }
```

Comment:
I agree this kind of expect is not a good experience

Reply:
Hmm... let's be clear, we received a broken on-chain config and there's no panic? I don't see how this is recoverable.

Reply:
Or maybe this would panic if we add a new config?